File: LEETCODE-PROBLEMS/26 - Remove Duplicates from Sorted - Easy/RemoveDuplicates.py
class RemoveDuplicates(object):
    
    # Clumsy approach
    def removeduplicates(self, nums):
        s = set()
        res = []
        for n in nums:
            s.add(n)
        for n in s:
            res.append(n)
        return res
    
    # Also known as runner technique
    def removeDuplicates_twoPointer(self, nums):
        i, j = 0, 0                 # initialize pointers i and j
        while j < len(nums):        
            if nums[j] != nums[i]:  # check if i and j of nums match
                i += 1              # move i right
                nums[i] = nums[j]   # swap i and j of nums
            j += 1                  # move j right
        return nums[:i + 1]         # return sliced to last unique element
        # return len(nums[:i + 1])  # return this for the number of unique elements
        

rd = RemoveDuplicates()
print(rd.removeDuplicates_twoPointer([1,1,2,3]))

File: LEETCODE-PROBLEMS/26 - Remove Duplicates from Sorted - Easy/RemoveDuplicates.js
/**
 * Removes duplicates of sorted input array
 * @param {number[]} nums input of sorted numbers
 * @returns {number[]} array of unique elements in nums
 */
var removeDuplicates = function(nums) {
    let i = 0, j = 0;
    while (j < nums.length) {
        if (nums[j] !== nums[i]) {
            i += 1;
            nums[i] = nums[j];
        } j += 1
    } return nums.slice(0, i + 1);
    // return nums.slice(0, i + 1).length
}